### ISSUE TITLE
Non-actionable errors from unixd on Debian system users: user uid and gid are not the same, this may be a security risk!

### ISSUE THREAD

**Issue by jinnatar** [kanidm/kanidm#3304]
### I did this
Started unixd 1.4.4 on a Debian bookworm server.

### I expected the following
Less seemingly fatal errors on startup.

### This happened instead
Quite a bit fatal seeming errors on startup:
> Dec 17 16:14:02 examplehost kanidm_unixd[758201]: 00000000-0000-0000-0000-000000000000 ERROR    🚨 [error]: user uid and gid are not the same, this may be a security risk! | name: systemd-network | uid: 101 | gid: 102
Dec 17 16:14:02 examplehost kanidm_unixd[758201]: 00000000-0000-0000-0000-000000000000 ERROR    🚨 [error]: user uid and gid are not the same, this may be a security risk! | name: systemd-resolve | uid: 102 | gid: 103
Dec 17 16:14:02 examplehost kanidm_unixd[758201]: 00000000-0000-0000-0000-000000000000 ERROR    🚨 [error]: user uid and gid are not the same, this may be a security risk! | name: systemd-timesync | uid: 104 | gid: 110

The above is just a small snapshot, in total there are 31 users that have this mismatch, all installed and thus users created via apt. All of them have an uid < 1000.

### Kanidm version details

- Output of `kanidm(d) version`: kanidm-unixd `1.4.4-202412030555+c3dbf83`
- Are you running it in a container? If so, which image/tag?: No
- If not a container, how'd you install it: unixd is from PPA, server from container.
- Operating System / Version (On Unix please post the output of `uname -a`): `Linux examplehost 6.1.0-28-amd64 #1 SMP PREEMPT_DYNAMIC Debian 6.1.119-1 (2024-11-22) x86_64 GNU/Linux`

### Any other comments

```shell
% > grep systemd-network /etc/{passwd,group}
/etc/passwd:systemd-network:x:101:102:systemd Network Management,,,:/run/systemd:/usr/sbin/nologin
/etc/group:systemd-network:x:102:
```

My issue here is that as far as I know, this behavior is completely normal and to be expected, i.e. it's exceedingly disruptive for an admin to "fix". Unixd probably shouldn't emit this error for uids under 1000 and instead assume the distro "knows what it's doing".

Some of the erroring entities also mismatch uid & gid very much on purpose, for example:
> uid=105(sshd) gid=65534(nogroup) groups=65534(nogroup)

**Comment by jinnatar** [comment]
.. Of course I notice commit eba8dff only after filing this, so this seems on track for a fix in 1.5.0/1.4.5 I guess. :-)

**Comment by Firstyear** [comment]
@jinnatar The problem is that distros "do not" actually know the risks, so the warnings are real. Generally it would be best to raise bug reports about it, but I guess that has a low chance to fix things. 

Anyway, yes, were going to be a bit more careful about these warnings in future.
